doc/source/whatsnew/v0.23.1.txt

```diff
@@ -23,7 +23,7 @@ New features
 Deprecations
 ~~~~~~~~~~~~
 
--
+- :func:`read_stata` and :class:`StataReader` have deprecated the ``encoding`` parameter. Stata files only support a single encoding and so this input has no effect. (:issue:`21244`)
```

Comment:
I know you say it didn't have any effect before, but still, we should not introduce deprecation warnings in a bug fix release. So would either move this to 0.24.0 or split the PR